synthesizer/program/src/resources/credits.aleo

```diff
@@ -106,573 +149,507 @@ record credits:
+            branch.eq r24 true to end;
+        // }
+        // If the delegated total is below 10M credits, continue to unbonding the validator.
+    //}
+
+    /* Unbond the validator */
     position unbond_validator;
-
-    /* Committee */
-
-    // Get the committee state for the specified validator.
-    get committee[r0] into r6;
-    // Decrement the stake for the specified validator.
-    sub r6.microcredits r1 into r7;
-
-    /* Bonded */
-
-    // Get the bond state for the validator, or fail if it does not exist.
-    get bonded[r0] into r8;
-    // Ensure that the validator matches in the bond state.
-    assert.eq r8.validator r0;
-    // Decrement the microcredits in the bond state.
-    sub r8.microcredits r1 into r9;
-
-    // Determine if the remaining bond is at least 10 million credits.
-    gte r9 10_000_000_000_000u64 into r10;
-
-    // If the remaining balance is at least 10 million credits, jump to the `decrement_validator` logic.
-    branch.eq r10 true to decrement_validator;
-    // If the remaining balance is less than 10 million credits, jump to the `remove_validator` logic.
-    branch.eq r10 false to remove_validator;
-
-    /*** Decrement Validator ***/
-
-    // Starts the `decrement_validator` logic.
-    position decrement_validator;
-
-    /* Committee */
-
-    // Construct the updated committee state.
-    cast r7 r6.is_open into r11 as committee_state;
-    // Update the committee state for the validator.
-    set r11 into committee[r0];
-
-    /* Bonded */
-
-    // Construct the updated bond state.
-    cast r0 r9 into r12 as bond_state;
-    // Update the bond state for the validator.
-    set r12 into bonded[r0];
-
-    /* Unbonding */
-
-    // Increment the microcredits in the unbond state.
-    add r3.microcredits r1 into r13;
-
-    // Construct the updated unbond state.
-    cast r13 r4 into r14 as unbond_state;
-    // Update the unbond state for the caller.
-    set r14 into unbonding[r0];
-
-    // Ends the `decrement_validator` logic.
-    branch.eq true true to end;
-
-    /*** Remove Validator ***/
-
-    // Starts the `remove_validator` logic.
+    // {
+        /* Setup Unbond Validator */
+        // Check if the committee contains the validator
+        contains committee[r5.validator] into r25;
+        // Delegator unbonding from a validator not in the committee
+        nor r11 r25 into r26;
+        // Jump to end if delegator unbonding from a validator not in the committee
+        branch.eq r26 true to end;
+
+        // Retrieve the committee state or fail if it doesn't exist.
+        get committee[r5.validator] into r27;
+
+        // Retrieve the bond state of the validator.
+        get bonded[r5.validator] into r28;
+
+        // Retrieve the total delegated balance for the validator.
+        get delegated[r5.validator] into r29;
+        /* End Setup Unbond Validator */
+
+        // Check if the current delegated amount is below 10M credits.
```

Comment:
Update comment to note this is only if the delegator caused this to drop below 10M above